traits.lua

```diff
@@ -1,36 +1,64 @@
 -- SPDX-License-Identifier: Apache-2.0
 -- SPDX-FileCopyrightText: 2025 Fundament Software SPC <https://fundament.software>
+
+local next, rawget, rawset, type = next, rawget, rawset, type
+
 ---@class LuaTrait
 ---@field declare_method fun(self: LuaTrait, methodname: string)
+---@field implement_method_on fun(self: LuaTrait, ty: Type, method_name: string, method: function)
 ---@field implement_on fun(self: LuaTrait, ty: Type, methods: { [string]: function })
 ---@field get fun(self: LuaTrait, ty: Type): { [string]: function }
 
 local trait_type_mt = {
 	__index = {
 		declare_method = function(self, methodname)
-			if self.methods[methodname] then
+			local self_methods = rawget(self, "methods")
+			if self_methods[methodname] then
 				error("trait " .. self.name .. " method " .. methodname .. " is already declared")
 			end
-			self.methods[methodname] = true
+			self_methods[methodname] = true
+		end,
+		implement_method_on = function(self, ty, method_name, method)
+			-- if type(ty) ~= "table" then
```

Comment:
is this supposed to be commented?

Reply:
That should probably be consistent between `implement_method_on` and `implement_method`. Does it really matter if we implement a method on a non-table? I don't think the trait system cares, and the `type()` call shows up in the profiler when it's present.

Reply:
Removing from both for consistency seems best in that case

@aiverson remember if there's a reason this has to only allow tables?

Reply:
Done, along with making `implement_method_on` safer.